### PR TITLE
Add basicAuth config and resilient config for Hub

### PR DESCRIPTION
**Pull request by chamil321**
> Add basic provider and config to both publisher and hub

> Add required resiliency config to the hub client

### CODE REVIEW

File: src/pubhub/pub_hub.bal [hunk 1]
```diff
@@ -8,13 +9,33 @@ import ballerinax/java.jdbc;
...
+http:BasicAuthHandler inboundBasicAuthHandler = new(new auth:InboundBasicAuthProvider());
+
 websub:WebSubHub webSubHub = startHubAndRegisterTopic();
 
-listener http:Listener httpListener = new (config:getAsInt("eclk.pub.port", 8181));
+listener http:Listener httpListener = new (config:getAsInt("eclk.pub.port", 8181), config = {
+    auth: {
+        authHandlers: [inboundBasicAuthHandler]
+    },
+    secureSocket: {
```
Comment (MaryamZi) [inline src/pubhub/pub_hub.bal:21]: If the keystore and/or truststore are not specified the logic here can fail right.

How about we move this to a function

```suggestion
    secureSocket: getSecureSocket()
```

where get secure socket will return `()` if neither keystore nor truststore are specified (i.e., config lookup returns ""). If only one of the two is specified, we can only set the relevant one and return. WDYT?

File: src/pubhub/pub_hub.bal [hunk 2]
```diff
@@ -102,7 +123,17 @@ function startHubAndRegisterTopic() returns websub:WebSubHub {
     mysqlstore:MySqlHubPersistenceStore persistenceStore = checkpanic new (subscriptionDb);
 
     websub:WebSubHub | websub:HubStartedUpError hubStartUpResult =
-        websub:startHub(new http:Listener(config:getAsInt("eclk.hub.port", 9090)),
+        websub:startHub(new http:Listener(config:getAsInt("eclk.hub.port", 9090), config = {
+                          auth: {
+                              authHandlers: [inboundBasicAuthHandler]
+                          },
+                          secureSocket: {
```
Comment (MaryamZi) [inline src/pubhub/pub_hub.bal:130]: same comment re: failure when keystore is not set.

File: src/pubhub/pub_hub.bal [hunk 3]
```diff
@@ -113,6 +144,14 @@ function startHubAndRegisterTopic() returns websub:WebSubHub {
                                 retryConfig: {
                                     count:  3,
                                     intervalInMillis: 5000
+                                },
+                                followRedirects: { enabled: true, maxCount: 5 },
```
Comment (MaryamZi) [inline src/pubhub/pub_hub.bal]: Shall we reformat this similar to the other configs? Values on new lines?

File: src/pubhub/pub_hub.bal [hunk 4]
```diff
@@ -113,6 +144,14 @@ function startHubAndRegisterTopic() returns websub:WebSubHub {
...
+                                timeoutInMillis: 5*60000, // Check
+                                secureSocket: {
```
Comment (MaryamZi) [inline src/pubhub/pub_hub.bal:153]: Same comment as above. We should be able to reuse the same method, passing the keys as arguments.

File: src/pubhub/sample-users.toml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[b7a.websub.hub.auth]
```
Comment (MaryamZi) [inline src/pubhub/sample-users.toml:2]: Shall we add a comment saying this is a test file.

File: src/pubhub/pub_hub.bal (hunk 1 above)
Comment (chamil321) [reply src/pubhub/pub_hub.bal:21]: No, You cant test without them as you need secure socket config to enable auth providers.
When you return () from the method that you specified, following will result.

```
Secure sockets have not been configured in order to enable auth providers. 
```

Comment (MaryamZi) [reply src/pubhub/pub_hub.bal:21]: Got it, didn't know the dependency with auth providers.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**